jina/executors/indexers/query/keyvalue.py

```diff
@@ -0,0 +1,49 @@
+from typing import Iterable
+
+from jina.executors.dump import import_metas
+from jina.executors.indexers.keyvalue import BinaryPbIndexer
+from jina.executors.indexers.query import BaseQueryIndexer
+
+
+class QueryBinaryPbIndexer(BinaryPbIndexer, BaseQueryIndexer):
+    """A write-once Key-value indexer."""
+
+    def load_dump(self, dump_path):
+        """Load the dump at the path
+
+        :param dump_path: the path of the dump"""
+        ids, metas = import_metas(dump_path, str(self.pea_id))
+        self.add(list(ids), list(metas))
+        self.write_handler.flush()
+        self.write_handler.close()
+        self.handler_mutex = False
+        self.is_handler_loaded = False
+        del self.write_handler
+        # warming up
+        self.query('someid')
```

Comment:
why is there a need for `query`?

Reply:
This is the queryindexer. Or do you mean the warming up? It's to prepare the `read_handler` to be in-memory (as much as it can in the case of the BinaryPb -- only the header is actually fully loaded). It's basically front-loading any processing needed further on for querying

Reply:
ah okey